optuna_integration/__init__.py

```diff
@@ -0,0 +1,110 @@
+    from optuna_integration.shap import ShapleyImportanceEvaluator
+    from optuna_integration.sklearn import OptunaSearchCV
+    from optuna_integration.skorch import SkorchPruningCallback
+    from optuna_integration.tensorboard import TensorBoardCallback
+    from optuna_integration.tensorflow import TensorFlowPruningHook
+    from optuna_integration.tfkeras import TFKerasPruningCallback
+else:
+
+    class _IntegrationModule(ModuleType):
+        """Module class that implements `optuna_integration` package.
+
+        This class applies lazy import under `optuna_integration`, where submodules are imported
+        when they are actually accessed. Otherwise, `import optuna` becomes much slower because it
+        imports all submodules and their dependencies (e.g., chainer, keras, lightgbm) all at once.
+        """
+
+        __file__ = globals()["__file__"]
+        __path__ = [os.path.dirname(__file__)]
+
+        _modules = set(_import_structure.keys())
+        _class_to_module = {}
+        for key, values in _import_structure.items():
+            for value in values:
+                _class_to_module[value] = key
+
+        def __getattr__(self, name: str) -> Any:
+            if name in self._modules:
+                value = self._get_module(name)
+            elif name in self._class_to_module.keys():
+                module = self._get_module(self._class_to_module[name])
+                value = getattr(module, name)
+            else:
+                raise AttributeError("module {} has no attribute {}".format(self.__name__, name))
+
+            setattr(self, name, value)
+            return value
+
+        def _get_module(self, module_name: str) -> ModuleType:
+            import importlib
+
+            try:
+                return importlib.import_module("." + module_name, self.__name__)
+            except ModuleNotFoundError:
+                raise ModuleNotFoundError(
+                    "Optuna's integration modules for third-party libraries have started "
+                    "migrating from Optuna itself to a package called `optuna-integration`. "
+                    "The module you are trying to use has already been migrated to "
+                    "`optuna-integration`. Please install the package by running "
+                    "`pip install optuna-integration`."
+                )
+
+    sys.modules[__name__] = _IntegrationModule(__name__)
+
+__all__ = [
+    "AllenNLPExecutor",
+    "AllenNLPPruningCallback",
+    "CatalystPruningCallback",
+    "CatBoostPruningCallback",
+    "ChainerMNStudy",
+    "ChainerPruningExtension",
+    "FastAIPruningCallback",
+    "FastAIV1PruningCallback",
+    "FastAIV2PruningCallback",
+    "KerasPruningCallback",
+    "MXNetPruningCallback",
+    "OptunaSearchCV",
+    "ShapleyImportanceEvaluator",
+    "SkorchPruningCallback",
+    "TensorBoardCallback",
+    "TensorFlowPruningHook",
+    "TFKerasPruningCallback",
+]
```

Comment:
```suggestion
        __all__ = [
            "AllenNLPExecutor",
            "AllenNLPPruningCallback",
            "CatalystPruningCallback",
            "CatBoostPruningCallback",
            "ChainerMNStudy",
            "ChainerPruningExtension",
            "FastAIPruningCallback",
            "FastAIV1PruningCallback",
            "FastAIV2PruningCallback",
            "KerasPruningCallback",
            "MXNetPruningCallback",
            "OptunaSearchCV",
            "ShapleyImportanceEvaluator",
            "SkorchPruningCallback",
            "TensorBoardCallback",
            "TensorFlowPruningHook",
            "TFKerasPruningCallback",
        ]

    sys.modules[__name__] = _IntegrationModule(__name__)

```
`__all__` should be defined inside _IntegrationModule class, because the module itself will be overwritten.